### PR TITLE
torch.view_as_complex fails with segfault for a zero dimensional tensor

### LINKED ISSUE #44061: Segmentation Fault: torch.view_as_complex fails with segfault for a zero dimensional tensor
`torch.view_as_complex` should throw an error for zero dimensional tensors.
```
>>> torch.view_as_complex(torch.tensor(2.0))
Segmentation fault (core dumped)
```

cc @ezyang @anjali411 @dylanbespalko

### PR DESCRIPTION
**Pull request by guol-fnst**
Fixes #44061

### CODE REVIEW

**Comment by guol-fnst** [comment]
b

**Comment by guol-fnst** [comment]
Hi @anjali411 
Could you please help to review this PR?
Thanks!

File: aten/src/ATen/native/ComplexHelper.h [hunk 1]
```diff
@@ -48,6 +48,7 @@ Tensor view_as_complex(const Tensor& self) {
     self.scalar_type() == kFloat || self.scalar_type() == kDouble || self.scalar_type() == kHalf,
     "view_as_complex is only supported for half, float and double tensors, but got a tensor of scalar type: ", self.scalar_type());
 
+  TORCH_CHECK(self.dim() != 0, "Dimension of tensor can't be zero");
```
Comment (anjali411) [inline aten/src/ATen/native/ComplexHelper.h:51]: maybe - `"Input tensor must have one or more dimensions."`

**Review by anjali411 — APPROVED** [review]
thanks for the fix @guol-fnst

File: aten/src/ATen/native/ComplexHelper.h (hunk 1 above)
Comment (anjali411) [reply aten/src/ATen/native/ComplexHelper.h:51]: alternately the check below can also be just modified to:
`TORCH_CHECK(self.dim() != 0 && new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");`

File: aten/src/ATen/native/ComplexHelper.h [hunk 2]
```diff
@@ -48,8 +48,9 @@ Tensor view_as_complex(const Tensor& self) {
     self.scalar_type() == kFloat || self.scalar_type() == kDouble || self.scalar_type() == kHalf,
     "view_as_complex is only supported for half, float and double tensors, but got a tensor of scalar type: ", self.scalar_type());
 
+  TORCH_CHECK(self.dim() != 0, "Input tensor must have one or more dimensions");
   auto new_sizes = self.sizes().vec();
-  TORCH_CHECK(new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");
+  TORCH_CHECK(self.dim() != 0 && new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");
```
Comment (VinodSKumar) [inline aten/src/ATen/native/ComplexHelper.h:53]: Redundant check for non zero dimension can be avoided, since it has already been checked above.

File: aten/src/ATen/native/ComplexHelper.h [hunk 3]
```diff
@@ -48,8 +48,9 @@ Tensor view_as_complex(const Tensor& self) {
     self.scalar_type() == kFloat || self.scalar_type() == kDouble || self.scalar_type() == kHalf,
     "view_as_complex is only supported for half, float and double tensors, but got a tensor of scalar type: ", self.scalar_type());
 
+  TORCH_CHECK(self.dim() != 0, "Input tensor must have one or more dimensions");
```
Comment (anjali411) [inline aten/src/ATen/native/ComplexHelper.h:51]: nit - we should remove this check since it's already being checked below

File: aten/src/ATen/native/ComplexHelper.h (hunk 2 above)
Comment (anjali411) [reply aten/src/ATen/native/ComplexHelper.h:53]: @guol-fnst yeah we should remove the check here

Comment (guol-fnst) [reply aten/src/ATen/native/ComplexHelper.h]: Yes, this check has benn removed
